src/Queue/QueueBase.php

```diff
@@ -0,0 +1,96 @@
+<?php
+
+/**
+ * @file
+ * Contains \Drupal\redis\Queue\QueueBase.
+ */
+
+namespace Drupal\redis\Queue;
+
+use Drupal\Core\Queue\ReliableQueueInterface;
+
+/**
+ * Redis queue implementation.
+ *
+ * @ingroup queue
+ */
+abstract class QueueBase implements ReliableQueueInterface {
```

Comment:
This is indeed reliable (I guess it is) since you used transactions, but beware of the way they work! And as I said upper, it's not shardable, so my opinion here is you have to make a choice here:
*   Convert this implementation using no MULTI blocks, case in which you cannot implement Drupal\Core\Queue\ReliableQueueInterface
*   Write another implementation and keep this one, but the other one would basically be a copy/paste without the MULTI blocks, which would not be Reliable
I'd prefer the later, both to be implemented, leaving the choice for the site admin which one to use depending on his environment.